doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -0,0 +1,237 @@
+Upon each boot the Secure Domain checks if the recovery flag is set.
+If it is set, the device enters recovery mode.
+If it isn't booting proceeds normally by running the root manifest.
+
+After entering the recovery mode it is verified if MPI configuration for the APP_RECOVERY is present.
+If it is, the APP_RECOVERY manifest is processed.
+It no MPI configuration is found the Secure Domain performs an attempt to process the normal manufacturer root manifest.
+This is needed, as the device might enter recovery mode if it is empty.
+The recovery flag is not cleared after flashing the firmware, but the device should proceed as if it would boot normally.
+
+This flow is shown in the following diagram:
+
+.. figure:: images/nrf54h20_suit_recovery_boot_path .png
+   :alt: Booting in recovery mode
+
+The role of the recovery application is to perform an update of the main application firmware, which does not differ from the normal SUIT update process.
+As soon as the update finishes successfully, the recovery flag is cleared and the device proceeds with normal operation.
+
+.. note::
+    The recovery application itself can only be updated from the main application - not when running the recovery application.
+
+.. _ug_nrf54h20_suit_recovery_default_fw:
+
+Using the default recovery firmware
+***********************************
+
+Nordic provides a default recovery firmware that can be used in the recovery process.
+This firmware uses Bluetooth LE and SMP as a transport.
+It is optimized for memory usage, currently using around 164 kB of MRAM (72 kB of application core and 92 kB radio core).
+
+.. caution::
+   The default recovery firmware does not support :ref:`updating from external flash memory <ug_nrf54h20_suit_external_memory>`.
+   This also means that the recovery firmware is not compatible with a main application that uses external flash for updates.
+
+To use the firmware:
+
+1. Create :file:`recovery.overlay` and :file:`recovery_hci_ipc.ovelay` files in the main application's :ref:`configuration_system_overview_sysbuild` directory.
+   These files must define the ``cpuapp_recovery_partition`` and ``cpurad_recovery_partition`` nodes respectively.
+   These partitions specify where the images for the recovery firmware are stored and cannot overlap with the main application partitions.
+   For reference, see the files in the :ref:`SUIT DFU on the nRF54H20 SoC <nrf54h_suit_sample>` sample located in :file:`samples/suit/smp_transfer`.
+
+#. Set the :kconfig:option:`SB_CONFIG_SUIT_BUILD_RECOVERY` sysbuild configuration option in the main application.
+   This will cause the recovery firmware to be built automatically as part of the main application build.
+
+#. :ref:`Program the main application firmware to the device <programming>`.
+This will automatically program both the main application and the recovery firmware to the device.
+
+#. To perform the recovery perform an update in the same way as described in :ref:`nrf54h_suit_sample`.
```

Comment:
Nit: To perform the recovery (_perform an_) update **follow** (_in_) the same **steps** (_way_) as described in ?

Reply:
Except for the nitpick (I reworded the sentence), I have also added the information which envelopes need to be used (not the root.suit envelope, but the app_recovery.suit envelope)